include/graaflib/algorithm/coloring/welsh_powell.tpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <graaflib/algorithm/coloring/welsh_powell.h>
+
+#include <unordered_map>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+namespace graaf::algorithm {
+
+template <typename GRAPH>
+std::unordered_map<vertex_id_t, int> welsh_powell_coloring(const GRAPH& graph) {
+    
+    using DegreeVertexPair = std::pair<int, vertex_id_t>;
+
+    // Step 1: Sort vertices by degree in descending order
+    std::vector<DegreeVertexPair> degree_vertex_pairs;
```

Comment:
[very minor]: In the rest of the code base we mainly use snake case. Also see our [c++ guidelines](https://github.com/bobluppes/graaf/wiki/guidelines).

```suggestion
    using degree_vertex_pair = std::pair<int, vertex_id_t>;

    // Step 1: Sort vertices by degree in descending order
    std::vector<degree_vertex_pair> degree_vertex_pairs;
```

Reply:
Done!!!